authors/apps/authentication/views.py

```diff
@@ -29,7 +29,7 @@ def post(self, request):
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
 
-class LoginAPIView(APIView):
+class LoginAPIView(GenericAPIView):
```

Comment:
Do we need to switch `GenericAPIView`? Can we not make swagger work with other `DRF Views`?

Reply:
I had read in a document that the serializer_class attribute works only with GenericAPIView. I tried using APIView but it did not render the contents of the swagger documentation until I used GenericApiView.

Reply:
If we take this approach we may end up having all our views extending `GenericAPIView`.

Reply:
Okay, Let me change the remaining view to extend GenericAPIView.

Reply:
@inno-asiimwe , what if i leave the remaining function with the class its extending that is `RetrieveUpdateAPIView` in order to not change the functionality of the endpoint.